roles/edpm_deploy_baremetal/tasks/main.yml

```diff
@@ -196,6 +196,20 @@
             "path": "/spec/nodeTemplate/ansible/ansibleVars/edpm_container_registry_insecure_registries",
             "value": [{{ content_provider_registry_ip }}:5001]}]'
 
+- name: Patch OpenStackDataPlaneNodeSet resource to update bootstrap packages
+  when: not cifmw_edpm_deploy_baremetal_dry_run
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.command:
+    cmd: >-
+      oc patch openstackdataplanenodeset openstack-edpm-ipam
+      -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }}
+      --type json
+      -p '[{"op": "add",
+            "path": "/spec/nodeTemplate/ansible/ansibleVars/edpm_bootstrap_command",
+            "value": "sudo dnf -y update"}]'
+
```

Comment:
can you please add some more context in the commit message. I see there was some discussion in https://github.com/openstack-k8s-operators/ci-framework/pull/936/files#r1425373973

which lead to this code being removed. What has since changed that we now want to add it back? 

Otherwise this looks correct/this is the code that was removed

Reply:
https://github.com/openstack-k8s-operators/edpm-ansible/blob/5ead70322381b4787bb735e27e0a11799f0c0549/roles/edpm_bootstrap/tasks/update_package.yml would be responsible for updating the package. But it is not called in this playbook https://github.com/openstack-k8s-operators/edpm-ansible/blob/5ead70322381b4787bb735e27e0a11799f0c0549/roles/edpm_bootstrap/tasks/bootstrap.yml (which runs the bootstrap-openstack-edpm-ipam pod).

It is safe to merge this pr right now to fix the broken job. I will do a proper fix on edpm-ansible side and revert back.

Reply:
Logs from failing job here: https://logserver.rdoproject.org/74/36274/46/check/periodic-podified-edpm-baremetal-master-ocp-crc/5621ca9/ci-framework-data/logs/quay-io-openstack-k8s-operators-openstack-must-gather-sha256-206e6f463855943ce5f128d1731718c7a7e0faf5bb2cc0fda0ddeb567abeb18d/namespaces/openstack/pods/bootstrap-openstack-edpm-ipam-ns4nj/logs/openstackansibleee.log
```
osp.edpm.edpm_bootstrap : Create swapfile if needed -------------------- 15.19s
osp.edpm.edpm_bootstrap : Gather services facts ------------------------- 7.88s
osp.edpm.edpm_kernel : Remove dracut-config-generic --------------------- 4.71s
osp.edpm.edpm_bootstrap : Install required packages to bootstrap EDPM --- 4.17s
osp.edpm.edpm_kernel : Disable KSM services ----------------------------- 3.61s
Gathering Facts --------------------------------------------------------- 3.28s
osp.edpm.edpm_bootstrap : Ensure packages are actually well installed --- 2.57s
osp.edpm.edpm_tuned : Enable tuned profile ------------------------------ 2.06s
osp.edpm.edpm_kernel : Render sysctl options ---------------------------- 2.00s
osp.edpm.edpm_tuned : Install tuned ------------------------------------- 1.80s
osp.edpm.edpm_users : Ensure user is present on the host [ qemu ] ------- 1.68s
osp.edpm.edpm_kernel : Write list of modules to load at boot ------------ 1.66s
osp.edpm.edpm_bootstrap : Set selinux state ----------------------------- 1.59s
osp.edpm.edpm_bootstrap : Deploy network-scripts required for deprecated network service --- 1.59s
osp.edpm.edpm_tuned : Enable and restart tuned -------------------------- 1.50s
osp.edpm.edpm_bootstrap : Ensure network service is enabled ------------- 1.49s
osp.edpm.edpm_tuned : Check Tuned Configuration file exists ------------- 1.45s
osp.edpm.edpm_users : Ensure group is present on the host [ qemu ] ------ 1.40s
osp.edpm.edpm_bootstrap : Enable openvswitch service if installed ------- 1.31s
osp.edpm.edpm_tuned : Check tuned active profile ------------------------ 1.29s
```

Reply:
@marios I have updated the commit message https://github.com/openstack-k8s-operators/ci-framework/pull/994#issue-2064713299 . Going to approve it to unblock promotion.

Reply:
Added https://github.com/openstack-k8s-operators/edpm-ansible/pull/530 pr to fix edpm-ansible